modules/seqexec/server/src/main/scala/seqexec/server/gmos/GmosControllerEpics.scala

```diff
@@ -234,15 +237,20 @@ object GmosControllerEpics extends GmosEncoders {
         //If disperser or order are set, force mode configuration
         s.foldMap(_ => s ++ List(CC.setDisperserMode(disperserMode0).some))
       case cfg.GmosDisperser.OrderN(d, o, w) =>
-        val s0: Option[IO[Unit]] = setDisperser(state, d)
+        val s0: Option[IO[Unit]] = setDisperser(state, d) // None means disperser is already in the right position
```

Comment:
I don't know how this works so bear with me here if this is a dumb question.  The `IO` for applying the configuration change makes sense to me but I'm wondering how you know it is already in the right position without checking, and if you check, would that not require being in `IO`?

Reply:
Just before applying the configuration the current configuration is read from the instrument and passed around. That is the first argument, `state`. `setDisperser` will produce a `Some[IO[_]]` only if the current disperser is different from the demanded one, effectively removing unnecessary interactions with the instrument.

Reply:
Why not return `IO.unit` in this case?